packages/esm-patient-immunizations-app/src/immunizations/immunization-history-card.component.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ErrorState, PatientChartPagination } from '@openmrs/esm-patient-common-lib';
+import { Table, TableRow, TableBody, TableCell, DataTableSkeleton } from '@carbon/react';
+import { formatDate, parseDate, usePagination } from '@openmrs/esm-framework';
+import styles from './immunization-history-card.scss';
+import { useImmunizations } from '../hooks/useImmunizations';
+
+interface ImmunizationHistoryCardProps {
+  patientUuid: string;
+}
+
+const ImmunizationHistoryCard: React.FC<ImmunizationHistoryCardProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const { data: immunizations, error, isLoading } = useImmunizations(patientUuid);
+  const headerTitle = t('immunizations', 'Immunizations');
+  const pageSize = 5;
+
+  const sortedImmunizations = useMemo(() => {
+    return (immunizations || []).map((immunization) => ({
+      ...immunization,
+      existingDoses: (immunization?.existingDoses || []).sort(
+        (a, b) => new Date(a?.occurrenceDateTime).getTime() - new Date(b?.occurrenceDateTime).getTime(),
+      ),
+    }));
+  }, [immunizations]);
+
+  const tableRows = useMemo(() => {
+    return (sortedImmunizations || []).map((immunization) => ({
+      id: immunization.vaccineUuid,
+      vaccine: immunization.vaccineName,
+      doses: immunization.existingDoses.map((dose, index) => (
+        <div key={index} className={styles.doseCell}>
+          {'Dose ' + dose.doseNumber}
+          <br />
+          {formatDate(parseDate(dose.occurrenceDateTime), { mode: 'standard', time: 'for today' })}
+        </div>
+      )),
+    }));
+  }, [sortedImmunizations]);
+
+  const { results: row, currentPage, goTo } = usePagination(tableRows || [], pageSize);
+
+  if (isLoading) {
+    return (
+      <div className={styles.widgetCard}>
+        <DataTableSkeleton role="progressbar" />
+      </div>
+    );
+  }
+
+  if (error) {
+    return <ErrorState error={error} headerTitle={headerTitle} />;
+  }
+
+  const headers = [
+    { key: 'vaccine', header: 'Vaccine' },
+    { key: 'doses', header: 'Doses' },
+  ];
+
+  return (
+    <div className={styles.widgetCard}>
+      <div className={styles.headerRow}>
+        {headers.map((header, index) => (
+          <div key={index} className={styles.headerCell}>
+            {header.header}
+          </div>
+        ))}
+      </div>
+
+      <div>
+        <Table size="xl" useZebraStyles={false} aria-label="Immunization History">
```

Comment:
```suggestion
        <Table size="xl" useZebraStyles={false} aria-label="Immunization History">
```
 aria-label={t('immunizationhistory', 'Immunization History')}
Also we can adjust size based on the current device layout type, ensuring better usability and readability on tablet devices.
```
const layout = useLayoutType();
 const isTablet = layout === 'tablet';
 size={isTablet ? 'lg' : 'sm'}
```